pilot/pkg/serviceregistry/external/conversion.go

```diff
@@ -147,18 +147,24 @@ func convertEndpoint(service *model.Service, servicePort *networking.Port,
 	tlsMode := model.GetTLSModeFromEndpointLabels(endpoint.Labels)
 
 	return &model.ServiceInstance{
-		Endpoint: model.NetworkEndpoint{
-			Address:     addr,
-			Family:      family,
-			Port:        int(instancePort),
-			ServicePort: convertPort(servicePort),
-			Network:     endpoint.Network,
-			Locality:    endpoint.Locality,
-			LbWeight:    endpoint.Weight,
+		Endpoint: &model.IstioEndpoint{
+			Hostname:        service.Hostname,
+			Address:         addr,
+			Family:          family,
+			EndpointPort:    instancePort,
+			ServicePortName: servicePort.Name,
+			Network:         endpoint.Network,
+			Locality:        endpoint.Locality,
+			LbWeight:        endpoint.Weight,
+			Labels:          endpoint.Labels,
+			TLSMode:         tlsMode,
+			Attributes: model.ServiceAttributes{
+				Name:      service.Attributes.Name,
```

Comment:
we should also nuke the name piece. Otherwise, we can't share the endpoint across different service objects. I also dont see where this is used.

Reply:
That's beyond the scope of this PR. We'd have to give this more thought.